microgateway_decorator/bin/compile

```diff
@@ -7,12 +7,95 @@ BUILD_DIR=$1
 CACHE_DIR=$2
 ENV_DIR=$3
 
+# make sure that the APIGEE_MICROGATEWAY_NODEJS_LOCAL_INSTALL variable is
+# true or false and if it is not set then assume that it is false.
+function check_nodejslocalinstall_env_var
+{
+  echo "checking APIGEE_MICROGATEWAY_NODEJS_LOCAL_INSTALL variable."
+  if [ "$APIGEE_MICROGATEWAY_NODEJS_LOCAL_INSTALL" != true ] && [ "$APIGEE_MICROGATEWAY_NODEJS_LOCAL_INSTALL" != false ]; then
+    #if local install is not true or false then check if it is not a zero length string
+    #a zero length string is ok and we will assume false
+    if [ ! -z "$APIGEE_MICROGATEWAY_NODEJS_LOCAL_INSTALL" ]; then
+      echo "APIGEE_MICROGATEWAY_NODEJS_LOCAL_INSTALL must be set to true or false." >&2
+      exit 1
+    fi
+  fi
+  echo "APIGEE_MICROGATEWAY_NODEJS_LOCAL_INSTALL is not set, so assume false."
+}
+
+function untar
+{
+  echo "untar"
+  NODEJS_FILENAME=$1
+  tar --touch --strip-components 1 -xzf $NODEJS_FILENAME -C $BUILD_DIR/tmp/node
+}
+
+function wget_and_untar
+{
+  NODE_WGET=$1
+  NODEJS_FILENAME=$2
+  wget_output=$(wget --tries=1 --timeout=5 --quiet "$NODE_WGET")
+  if [ $? -ne 0 ]; then
+    echo "Node.js was not downloaded successfully from $NODE_WGET." >&2
+    exit 1
+  fi
+  untar $NODEJS_FILENAME
+  rm $NODEJS_FILENAME
+}
+
+# checks various combinations of the filename and the url and displays an error message
+# if the URL is valid protocol (http:// or https://)
+# or the url is missing and the file name is present
+# or the url is present and the file name is missing
+function check_nodejs_filename_and_url {
+  echo "Checking APIGEE_MICROGATEWAY_NODEJS_FILENAME and APIGEE_MICROGATEWAY_NODEJS_URL environment variables."
+  # check if the value is set to false or if it is zero length then assume false.
+  if [ "$APIGEE_MICROGATEWAY_NODEJS_LOCAL_INSTALL" == false ] || [ -z "$APIGEE_MICROGATEWAY_NODEJS_LOCAL_INSTALL" ]; then
+    echo "APIGEE_MICROGATEWAY_NODEJS_LOCAL_INSTALL is either set to false or assumed false."
+    if [ "$APIGEE_MICROGATEWAY_NODEJS_URL" != "" ] && [ "$APIGEE_MICROGATEWAY_NODEJS_FILENAME" == "" ]; then
```

Comment:
Since we are using `wget` we should be able to parse out the file name using `basename` from the `APIGEE_MICROGATEWAY_NODEJS_URL`, so we can get rid of the need for `APIGEE_MICROGATEWAY_NODEJS_FILENAME`

Reply:
We still need APIGEE_MICROGATEWAY_NODEJS_FILENAME in the event they want to install Node.js from the lib directory.  This var will also signal that this is a local install. But I will update the code to pull the file name from the URL instead.
filename=$(basename "$url")